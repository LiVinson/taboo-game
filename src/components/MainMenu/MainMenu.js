import React from 'react'
import PropTypes from 'prop-types'
import { TabooCard } from 'components/shared/TabooCard'
import TextLink from 'components/shared/TextLink'


const MainMenu = ({ match }) => {
	const { path } = match

	const cardInfo = {
		tabooWord: 'Menu',
		list: [
			<TextLink to={`${path}/create`} text={'Create New Game'} />,
			<TextLink to={`${path}/join`} text={'Join Game'} />,
			<TextLink to={`${path}/rules`} text={'How to Play'} />,
			<TextLink to={`${path}/submit`} text={'Submit a Card'} />,
		],
	}
	return <TabooCard {...cardInfo} />
}

MainMenu.propTypes = {
	match: PropTypes.object.isRequired,
}

export default MainMenu
