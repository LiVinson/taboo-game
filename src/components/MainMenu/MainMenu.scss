
    //Add styles to links once Main Menu is converted to styled components
    &__list-item:after {
        content:"";
        display:block;
        width: 10rem;
        margin: 0 auto;
        padding-top:1rem;
        border-bottom: #C4C4C4 2px solid; 
    }

    &__link:link,
    &__link:visited {
        color: var(--color-primary);
        text-decoration: none;
        text-shadow: 0px .4rem .4rem rgba(0,0,0,.7);
    }

    &__link:focus {
        text-shadow: 0px .2rem .4rem rgba(0,0,0,.7);
    }
