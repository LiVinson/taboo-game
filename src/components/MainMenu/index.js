import MainMenu from "./MainMenu"

export default MainMenu;