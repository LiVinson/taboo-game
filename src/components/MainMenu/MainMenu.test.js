import React from "react"
import { shallow } from "enzyme"
import renderer from "react-test-renderer"
import MainMenu from "./MainMenu"
import { StaticRouter } from "react-router-dom"
//Need static router to prevent react-router errors
it("renders correctly", () => {
  const props = {
    match: {
      path: "/home",
    },
  }

  const MainMenuComponent = renderer
    .create(
      <StaticRouter>
        <MainMenu {...props} />
      </StaticRouter>
    )
    .toJSON()

  expect(MainMenuComponent).toMatchSnapshot()
})
