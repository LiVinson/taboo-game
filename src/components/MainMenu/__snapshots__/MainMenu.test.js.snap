// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="taboo-card taboo-card--home"
>
  <div
    className="taboo-card__word-container"
  >
    <p
      className="taboo-card__word"
    >
      Menu
    </p>
  </div>
  <div
    className="taboo-card__list-container"
  >
    <ul
      className="taboo-card__list"
    >
      <li
        className="taboo-card__list-item"
      >
        <a
          aria-current={null}
          className="taboo-card__link"
          href="/home/create"
          onClick={[Function]}
        >
          Create New Game
        </a>
      </li>
      <li
        className="taboo-card__list-item"
      >
        <a
          aria-current={null}
          className="taboo-card__link"
          href="/home/join"
          onClick={[Function]}
        >
          Join Game
        </a>
      </li>
      <li
        className="taboo-card__list-item"
      >
        <a
          aria-current={null}
          className="taboo-card__link"
          href="/home/rules/0"
          onClick={[Function]}
        >
          How to Play
        </a>
      </li>
      <li
        className="taboo-card__list-item"
      >
        Submit a Card
      </li>
    </ul>
  </div>
</div>
`;
